build/parser.rs

```diff
@@ -339,12 +923,31 @@ pub fn parse_profile(file: &mut Read) -> MavProfile {
                             match attr.name.local_name.clone().as_ref() {
                                 "name" => {
                                     field.name = attr.value.clone();
+                                    if field.name == "type" {
+                                        field.name = "mavtype".to_string();
+                                    }
                                 }
                                 "type" => {
-                                    field.mavtype = parse_type(&attr.value).unwrap();
+                                    field.mavtype = MavType::parse_type(&attr.value).unwrap();
                                 }
                                 "enum" => {
-                                    field.enumtype = Some(attr.value.clone());
+                                    field.enumtype = Some(
+                                        attr.value
+                                            .clone()
+                                            .split("_")
+                                            .map(|x| x.to_lowercase())
+                                            .map(|x| {
+                                                let mut v: Vec<char> = x.chars().collect();
+                                                v[0] = v[0].to_uppercase().nth(0).unwrap();
+                                                v.into_iter().collect()
+                                            })
+                                            .collect::<Vec<String>>()
+                                            .join(""),
+                                    );
```

Comment:
This expression is duplicated twice above (one commented). Make it a function instead?